momento-sdk/src/main/java/momento/sdk/responses/topic/TopicMessage.java

```diff
@@ -0,0 +1,66 @@
+package momento.sdk.responses.topic;
+
+import grpc.cache_client.pubsub._TopicValue;
+import java.util.Arrays;
+import momento.sdk.exceptions.SdkException;
+
+public interface TopicMessage {
+
+  class Text implements TopicMessage {
+    private final String value;
+    private final String tokenId;
+
+    public Text(_TopicValue topicValue, String tokenId) {
+      this.value = topicValue.getText();
+      this.tokenId = tokenId;
+    }
+
+    public String getValue() {
+      return value;
+    }
+
+    public String getTokenId() {
+      return tokenId;
+    }
+
+    @Override
+    public String toString() {
+      return "Text{" + "value='" + value + '\'' + ", tokenId='" + tokenId + '\'' + '}';
+    }
+  }
+
+  class Binary implements TopicMessage {
+    private final byte[] value;
+    private final String tokenId;
+
+    public Binary(_TopicValue topicValue, String tokenId) {
+      this.value = topicValue.getBinary().toByteArray();
+      this.tokenId = tokenId;
+    }
+
+    public byte[] getValue() {
+      return value;
+    }
+
+    public String getTokenId() {
+      return tokenId;
+    }
+
+    @Override
+    public String toString() {
+      return "Binary{" + "value=" + Arrays.toString(value) + ", tokenId='" + tokenId + '\'' + '}';
```

Comment:
that `Arrays.toString` may not produce a useful representation of the data, and we may need to truncate these so that they don't blow out the console/logs for large items, but we can tweak that later once we have played with it more to see how it feels.